src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -692,9 +692,6 @@ private static unsafe int AlpnServerSelectCallback(IntPtr ssl, byte** outp, byte
                 return Ssl.SSL_TLSEXT_ERR_ALERT_FATAL;
             }
 
-            // reset application data to avoid dangling pointer.
-            Ssl.SslSetData(ssl, IntPtr.Zero);
```

Comment:
I'm wondering if we should move `AlpnHandle.Free()` to `ReleaseHandle`. It is now in `Dispose` and I'm wondering if something can `Dispose` the handle in just wrong time.

Reply:
Moved